controlplane/kubeadm/api/v1beta1/kubeadm_control_plane_webhook.go

```diff
@@ -616,6 +620,24 @@ func (in *KubeadmControlPlane) validateVersion(previousVersion string) (allErrs
 		)
 	}
 
+	// Block if imageRepository is not set (i.e. the default registry should be used),
```

Comment:
Let's add a comment on why we are doing this:
The Kubernetes ecosystem has been requested to move users to the new registry due to cost issues;
This validation enforces this, by forcing the users to upgrade to kubeadm versions with the new registry
NOTE: This is only for users relying on community maintained registry.
NOTE: Pinning to the upstream registry is not recommended because it could lead to issues given how the migration has been implemented in kubeadm.